src/Generator/Generators/CSharp/CSharpTextTemplate.cs

```diff
@@ -2035,6 +2035,13 @@ public void GenerateMethod(Method method, Class @class)
             if (method.IsPure)
                 Write("abstract ");
 
+            if(method.SynthKind == FunctionSynthKind.DefaultValueOverload)
+            {
+                var needUnsafe = method.Parameters.Any(p => p.Kind == ParameterKind.Regular && p.Ignore
+                             && p.Type.IsPointerToPrimitiveType() && p.Usage == ParameterUsage.InOut && p.HasDefaultValue);
+                Write("unsafe ");
```

Comment:
I read and understood your explanation. However, I removed all "unsafe" from your generated code and it still compiled and worked properly. If you really need this, please add a test that exposes the need. Otherwise please remove it.
This is the last major problem with this pull. Once you're done with it, we'll merge even if you decide to ignore the other minor remarks.